mlflow/transformers/__init__.py

```diff
@@ -996,6 +1000,90 @@ def load_model(
     return _load_model(local_model_path, flavor_config, return_type, device, **kwargs)
 
 
+def persist_pretrained_model(model_uri: str) -> None:
+    """
+    Persist Transformers pretrained model weights to the artifacts of the specified model_uri.
+    This API is primary for updating an MLflow Model that was logged or saved with setting
+    save_pretrained=False. Such model cannot be registered to the Model Registry, due to the
+    lack of actual pretrained model weights in the artifacts but only the reference to the
+    HuggingFace Hub repository. This API will download the model weights from the HuggingFace
+    Hub repository and save them in the artifacts of the given model_uri, so that the model
+    can be registered to the Model Registry.
+
+    Args:
+        model_uri: The URI of the existing MLflow Model of the Transformers flavor.
+            It must be logged/saved with save_pretrained=False.
+
+    Examples:
+
+    .. code-block:: python
+
+        import mlflow
+
+        # Saving a model with save_pretrained=False
+        with mlflow.start_run() as run:
+            model = pipeline("question-answering", "csarron/mobilebert-uncased-squad-v2")
+            mlflow.transformers.log_model(
+                transformers_model=model, artifact_path="pipeline", save_pretrained=False
+            )
+
+        # The model cannot be registered to the Model Registry as it is
+        try:
+            mlflow.register_model(f"runs:/{run.info.run_id}/pipeline", "qa_pipeline")
+        except MlflowException as e:
+            print(e.message)
+
+        # Use this API to persist the pretrained model weights
+        mlflow.transformers.persist_pretrained_model(f"runs:/{run.info.run_id}/pipeline")
+
+        # Now the model can be registered to the Model Registry
+        mlflow.register_model(f"runs:/{run.info.run_id}/pipeline", "qa_pipeline")
+    """
```

Comment:
Can we make this a no-op if the weights have already been loaded and are present in the directory? Since this is a public API, a user might call this function multiple times by accident and we only want a single copy of the weights. We can inform the user if the model weights are already loaded and let them know in a pleasant way that they've already been loaded and that they're "good to go" :)

Reply:
I see that we do that in `update_flavor_conf_to_persist_pretrained_model`, but is there a quick way to validate by listing the artifacts in the artifact store before downloading anything from the artifact store to local?

Reply:
Good idea! I think we can do that by checking `model` file exists or not in the artifact list. One concern is that the reliance of that directory name is not super robust (like in older version of MLflow we didn't have that instead `pipeline` directory), but I think it's fine as long as the worst case scenario is just unnecessary download.